File: NetworkPass-master/NetWorkPassServer/src/NetWorkPassServer.Application/Dtos/DeviesDtos/DeviceRealtimeDto.cs
using NetWorkPassServer.Domain.Devices;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NetWorkPassServer.Application.Dtos.DeviesDtos
{
    public sealed record DeviceRealtimeDto(
     Guid Id,

     Guid BranchId,

     string Name,

     DeviceStatus Status,

     long? PingLatency,

     double? CpuUsage,

     double? MemoryUsage,

     double? Temperature,

     DateTime? LastSeenAt
 );
}

File: NetworkPass-master/NetWorkPassClient/apps/Admin/src/Device/data-access/models/DeviceRealtimeDto.ts
export interface DeviceRealtimeDto {

  id: string;

  branchId: string;

  name: string;

  status: number;

  pingLatency: number | null;

  cpuUsage: number | null;

  memoryUsage: number | null;

  temperature: number | null;

  lastSeenAt: string | null;
}
